packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain/SourceNetworkBox.tsx

```diff
@@ -159,19 +178,22 @@ export function SourceNetworkBox({
             maxButtonOnClick={maxButtonOnClick}
             errorMessage={errorMessages?.inputAmount1}
             value={isMaxAmount ? '' : amount}
-            onChange={e => setAmount(e.target.value)}
+            onChange={handleAmountChange}
+            maxAmount={maxAmount}
+            isMaxAmount={isMaxAmount}
+            decimals={decimals}
           />
 
           {isBatchTransferSupported && (
             <TransferPanelMainInput
               maxButtonOnClick={amount2MaxButtonOnClick}
               errorMessage={errorMessages?.inputAmount2}
               value={amount2}
-              onChange={e => setAmount2(e.target.value)}
-              tokenButtonOptions={{
-                symbol: nativeCurrency.symbol,
-                disabled: true
-              }}
+              onChange={handleAmount2Change}
+              tokenButtonOptions={tokenButtonOptions}
+              maxAmount={maxAmount2}
+              isMaxAmount={isMaxAmount2}
+              decimals={nativeCurrency.decimals}
```

Comment:
this should be the native currency on the Parent Chain but this `nativeCurrency` is the child chain's

but as we know it'll always be 18, i think we can hardcode 18 instead

please check with @brtkx as he's working on the batch feature

Reply:
It will be native currency on the child chain, you can't send ETH to L3 custom gas token chain. And because we send L3 native currency from L2, it may be different than 18 decimals. So how it is now is correct.